app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,28 +123,178 @@ class PopupView : FrameLayout {
+                                    ee
+                                )
+                            )
+                        }
+                    }
+                }
+            } catch (ie: InterruptedException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ie
+                    )
                 )
-            )
+            } catch (ee: ExecutionException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ee
+                    )
+                )
+            }
         }
-        return popupDisplayFieldsWithValues
+    }
+
+    /**
+     * Checks for errors in FeatureEditResults obtained from applying edits to service feature table.
+     */
+    private fun checkEditResultForErrors(featureEditResults: List<FeatureEditResult>): Boolean {
+        var errorEncountered = false
+        for (result in featureEditResults) {
+            if (result.hasCompletedWithErrors()) {
+                errorEncountered = true
+                break
+            }
+        }
+        return errorEncountered
+    }
+
+
+    /**
+     * Updates the given popup field value with the appropriately cast string value.
+     *
+     * @param manager - PopupManager
+     * @param field   - PopupField
+     * @param value   - String
+     * @return - ArcGISRuntimeException if there's a problem updating the field value,
+     * otherwise null.
+     */
+    private fun updateValue(
+        manager: PopupManager,
+        field: PopupField?,
+        value: String?
+    ): ArcGISRuntimeException? {
+        var error: ArcGISRuntimeException? = null
+
+        when (manager.getFieldType(field)) {
+            Field.Type.SHORT -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toShort(), field)
+            }
+            Field.Type.INTEGER -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toInt(), field)
+            }
+            Field.Type.FLOAT -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toFloat(), field)
+            }
+            Field.Type.DOUBLE -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toDouble(), field)
+            }
+            Field.Type.DATE -> {
```

Comment:
you mentioned in slack this PR doesn't handle Calendar input. Is this case intentional?

Reply:
This here just updates the local geodatabase with the new values that the user enters. We are still not popping a calendar to help user enter dates via a calendar prompt...